released/discovery_center/mission_4356/main.tf

```diff
@@ -0,0 +1,135 @@
+###############################################################################################
+# Setup of names in accordance to naming convention
+###############################################################################################
+resource "random_uuid" "uuid" {}
+
+locals {
+  random_uuid               = random_uuid.uuid.result
+  project_subaccount_domain = lower(replace("mission-4172-${local.random_uuid}", "_", "-"))
+  project_subaccount_cf_org = substr(replace("${local.project_subaccount_domain}", "-", ""), 0, 32)
+}
+
+###############################################################################################
+# Creation of subaccount
+###############################################################################################
+resource "btp_subaccount" "project" {
+  name      = var.subaccount_name
+  subdomain = local.project_subaccount_domain
+  region    = lower(var.region)
+}
+
+###############################################################################################
+# Assignment of users as sub account administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-admins" {
+  for_each             = toset("${var.subaccount_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Administrator"
+  user_name            = each.value
+}
+
+###############################################################################################
+# Assignment of users as sub account service administrators
+###############################################################################################
+resource "btp_subaccount_role_collection_assignment" "subaccount-service-admins" {
+  for_each             = toset("${var.subaccount_service_admins}")
+  subaccount_id        = btp_subaccount.project.id
+  role_collection_name = "Subaccount Service Administrator"
+  user_name            = each.value
+}
+
+######################################################################
+# Creation of Cloud Foundry environment
+######################################################################
+resource "btp_subaccount_environment_instance" "cf" {
+  subaccount_id    = btp_subaccount.project.id
+  name             = local.project_subaccount_cf_org
+  environment_type = "cloudfoundry"
+  service_name     = "cloudfoundry"
+  plan_name        = "standard"
+  landscape_label  = var.cf_environment_label
+  parameters = jsonencode({
+    instance_name = local.project_subaccount_cf_org
+  })
+}
+
+######################################################################
+# Entitlement of all services
+######################################################################
+resource "btp_subaccount_entitlement" "name" {
+  depends_on = [btp_subaccount.project]
```

Comment:
depends on can be removed here; is covered by reference to subaccount ID

Reply:
This is fixed now.